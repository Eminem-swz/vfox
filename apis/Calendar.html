<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Calendar 日历选择 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.82adce24.css">
    <link rel="modulepreload" href="/vfox/assets/Home.472bd3cf.js">
    <link rel="modulepreload" href="/vfox/assets/app.9dc78301.js">
    <link rel="modulepreload" href="/vfox/assets/apis_Calendar.md.73c56524.lean.js">
    
    <meta name="twitter:title" content="Calendar 日历选择 | Vfox for Vue3">
  <meta property="og:title" content="Calendar 日历选择 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-675d8756><div class="sidebar-button" data-v-675d8756><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-675d8756 data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-675d8756></div><div class="nav" data-v-675d8756><nav class="nav-links" data-v-675d8756 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-83e92a68><nav class="nav-links nav" data-v-83e92a68 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-83e92a68><!--[--><li class="sidebar-link"><p class="sidebar-link-item">通知类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Notify">Notify</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">预览类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Preview">Preview</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">模态框类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Dialog">Dialog</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">抽屉类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/apis/Calendar">Calendar</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#showcalendar-object">showCalendar(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#object">object</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#object-success-回调参数">object.success 回调参数</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage">Usage</a><!----></li></ul></li></ul></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">气泡类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/PopMenu">PopMenu</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div><iframe style="display:none;width:375px;height:667px;margin-top:1.5rem;" src="https://godxiaoji.github.io/vfox/demo/#/" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="calendar-日历选择" tabindex="-1">Calendar 日历选择 <a class="header-anchor" href="#calendar-日历选择" aria-hidden="true">#</a></h1><h2 id="showcalendar-object" tabindex="-1">showCalendar(object) <a class="header-anchor" href="#showcalendar-object" aria-hidden="true">#</a></h2><p>日历选择弹窗。</p><h3 id="object" tabindex="-1">object <a class="header-anchor" href="#object" aria-hidden="true">#</a></h3><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>mode</td><td>string</td><td>&#39;single&#39;</td><td>否</td><td>模式：<code>single</code> 表示选择一天，<code>range</code> 选择一个日期区间</td></tr><tr><td>title</td><td>string</td><td></td><td>否</td><td>弹窗标题</td></tr><tr><td>value</td><td>Date/Date[]</td><td>[]</td><td>否</td><td>默认选择值，<code>range</code> 模式下需要提供两个</td></tr><tr><td>minDate</td><td>Date</td><td>当前日期</td><td>否</td><td>可选最小值</td></tr><tr><td>maxDate</td><td>Date</td><td>当前日期的六个月后</td><td>否</td><td>可选最大值</td></tr><tr><td>allowSameDay</td><td>boolean</td><td>false</td><td>否</td><td><code>range</code> 模式生效，设置开始结束时间是否可以同一天</td></tr><tr><td>maxRange</td><td>number</td><td>Infinity</td><td>否</td><td><code>range</code> 模式生效，选择区间的最长天数</td></tr><tr><td>dayHandler</td><td>Function</td><td></td><td>否</td><td>日历每个日期处理函数，参考 <a href="./../components/Calendar.html#dayhandler-函数">Calendar dayHandler</a></td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功（在用户做出选择后，如取消，确定）的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table><h3 id="object-success-回调参数" tabindex="-1">object.success 回调参数 <a class="header-anchor" href="#object-success-回调参数" aria-hidden="true">#</a></h3><table><thead><tr><th>属性</th><th>类型</th><th>说明</th></tr></thead><tbody><tr><td>cancel?</td><td>boolean</td><td>为 true 时，表示取消</td></tr><tr><td>confirm?</td><td>boolean</td><td>为 true 时，表示点击了确定，此时返回 <code>detail</code></td></tr><tr><td>detail?.value</td><td>Date[]</td><td>开始 1 个 Date 实例 或 开始和结束 2 个 Date 实例</td></tr><tr><td>detail?.valueArray</td><td>number[][]</td><td>如：[[2021, 5, 1], [2021, 5, 30]]</td></tr><tr><td>detail?.formatted</td><td>string</td><td>05-21 ~ 05-22</td></tr><tr><td>detail?.rangeCount</td><td>number</td><td>选择区间持续的天数（含首尾）</td></tr></tbody></table><h3 id="usage" tabindex="-1">Usage <a class="header-anchor" href="#usage" aria-hidden="true">#</a></h3><p>具体调用方式可以参考<a href="./../guide/import.html#api-调用">API 调用</a>。</p><div class="language-JavaScript"><pre><code><span class="token function">showCalendar</span><span class="token punctuation">(</span><span class="token punctuation">{</span>
  <span class="token literal-property property">type</span><span class="token operator">:</span> <span class="token string">&#39;range&#39;</span><span class="token punctuation">,</span>
  <span class="token literal-property property">showClose</span><span class="token operator">:</span> <span class="token boolean">true</span><span class="token punctuation">,</span>
  <span class="token function-variable function">success</span><span class="token operator">:</span> <span class="token punctuation">(</span><span class="token parameter"><span class="token punctuation">{</span> confirm<span class="token punctuation">,</span> cancel<span class="token punctuation">,</span> detail <span class="token punctuation">}</span></span><span class="token punctuation">)</span> <span class="token operator">=&gt;</span> <span class="token punctuation">{</span>
    <span class="token operator">...</span>
  <span class="token punctuation">}</span>
<span class="token punctuation">}</span><span class="token punctuation">)</span>
</code></pre></div></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><a class="link" href="/vfox/apis/Cascader" data-v-38ede35f><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-38ede35f><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-38ede35f>Cascader</span></a></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/apis/Popover" data-v-38ede35f><span class="text" data-v-38ede35f>Popover</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_actionsheet.md\":\"6e759880\",\"apis_calendar.md\":\"73c56524\",\"apis_cascader.md\":\"6de70566\",\"apis_datepicker.md\":\"e6503750\",\"apis_dialog.md\":\"6964068d\",\"apis_notify.md\":\"9d557c5d\",\"apis_picker.md\":\"6ad5b8f1\",\"apis_popdialog.md\":\"d4c2aefa\",\"apis_popmenu.md\":\"a0c9cfbe\",\"apis_popover.md\":\"a3645836\",\"apis_preview.md\":\"8b35e969\",\"apis_toast.md\":\"7568137d\",\"apis_index.md\":\"b1f704ad\",\"components_actionsheet.md\":\"52d22cba\",\"components_activityindicator.md\":\"b8a223e7\",\"components_avatar.md\":\"011d3fc7\",\"components_backtop.md\":\"e4e994f5\",\"components_badge.md\":\"028a7d96\",\"components_button.md\":\"8bf05d38\",\"components_calendar.md\":\"a1d6715d\",\"components_cascader.md\":\"66e4f298\",\"components_cell.md\":\"4ea2f836\",\"components_checkbox.md\":\"47b2c132\",\"components_circleprogress.md\":\"e5fcacc5\",\"components_collapse.md\":\"b43c15e1\",\"components_copy.md\":\"9e2aafef\",\"components_countdown.md\":\"1269c173\",\"components_countup.md\":\"f9854800\",\"components_datepicker.md\":\"87f0a811\",\"components_dialog.md\":\"25aab6d4\",\"components_divider.md\":\"d273a4eb\",\"components_drawer.md\":\"95ea9b04\",\"components_dropdown.md\":\"af86545f\",\"components_empty.md\":\"ca6616bb\",\"components_fixed.md\":\"31e38fdf\",\"components_flatlist.md\":\"f7a97393\",\"components_form.md\":\"3e13777e\",\"components_group.md\":\"ba614409\",\"components_icon.md\":\"c13bdbb3\",\"components_image.md\":\"cccd71f1\",\"components_imagepreview.md\":\"0139cb97\",\"components_imageuploader.md\":\"2c5b4d8c\",\"components_indexview.md\":\"b8b8cfb8\",\"components_input.md\":\"f8e72d81\",\"components_layout.md\":\"cabaf94c\",\"components_loadmore.md\":\"06109345\",\"components_modal.md\":\"872e1dd2\",\"components_navbar.md\":\"4750069d\",\"components_noticebar.md\":\"7d2b7c72\",\"components_notify.md\":\"b9eab969\",\"components_numberkeyboard.md\":\"896a2a73\",\"components_order.md\":\"4e95c01a\",\"components_pagination.md\":\"e4f0475d\",\"components_picker.md\":\"7af0057b\",\"components_popdialog.md\":\"d934a048\",\"components_popmenu.md\":\"2ce4cf62\",\"components_popover.md\":\"5ca63491\",\"components_price.md\":\"9799d29b\",\"components_progress.md\":\"b28c8651\",\"components_radio.md\":\"421c7a10\",\"components_range.md\":\"727c91aa\",\"components_rate.md\":\"a065896e\",\"components_result.md\":\"30cd9913\",\"components_scrolltab.md\":\"e12e16c3\",\"components_scrollview.md\":\"950370c9\",\"components_searchbar.md\":\"856dffb5\",\"components_sidetab.md\":\"ee102926\",\"components_skeleton.md\":\"821db3fc\",\"components_slider.md\":\"d7006cc7\",\"components_stepper.md\":\"49f3b7c2\",\"components_steps.md\":\"877b5ea0\",\"components_sticky.md\":\"16f4fc4c\",\"components_stopwatch.md\":\"e3735e07\",\"components_swipecell.md\":\"4eb6b4bb\",\"components_swiper.md\":\"586c4b1f\",\"components_switch.md\":\"9c810987\",\"components_tab.md\":\"40fea87a\",\"components_tabbar.md\":\"6641be22\",\"components_tabview.md\":\"5ddcafa6\",\"components_tag.md\":\"9e1e77f3\",\"components_timeago.md\":\"f342d517\",\"components_timeline.md\":\"0b56084b\",\"components_toast.md\":\"b942008b\",\"components_index.md\":\"8bc2015a\",\"design_color.md\":\"b30ef7ca\",\"design_index.md\":\"7ede16dc\",\"guide_dark.md\":\"01b89ba9\",\"guide_import.md\":\"6af40c44\",\"guide_locale.md\":\"a67725ac\",\"guide_theme.md\":\"a168b5c6\",\"index.md\":\"d0d3ac39\"}")</script>
    <script type="module" async src="/vfox/assets/app.9dc78301.js"></script>
    
  </body>
</html>