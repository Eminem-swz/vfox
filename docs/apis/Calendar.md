# Calendar 日历选择

## showCalendar(object)

日历选择弹窗。

### object

| 属性         | 类型        | 默认值             | 必填 | 说明                                                                                        |
| ------------ | ----------- | ------------------ | ---- | ------------------------------------------------------------------------------------------- |
| mode         | string      | 'single'           | 否   | 模式：`single` 表示选择一天，`range` 选择一个日期区间                                       |
| title        | string      |                    | 否   | 弹窗标题                                                                                    |
| value        | Date/Date[] | []                 | 否   | 默认选择值，`range` 模式下需要提供两个                                                      |
| minDate      | Date        | 当前日期           | 否   | 可选最小值                                                                                  |
| maxDate      | Date        | 当前日期的六个月后 | 否   | 可选最大值                                                                                  |
| allowSameDay | boolean     | false              | 否   | `range` 模式生效，设置开始结束时间是否可以同一天                                            |
| maxRange     | number      | Infinity           | 否   | `range` 模式生效，选择区间的最长天数                                                        |
| dayHandler   | Function    |                    | 否   | 日历每个日期处理函数，参考 [Calendar dayHandler](../components/Calendar.md#dayhandler-函数) |
| success      | function    |                    | 否   | 接口调用成功（在用户做出选择后，如取消，确定）的回调函数                                    |
| fail         | function    |                    | 否   | 接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）                    |
| complete     | function    |                    | 否   | 弹窗关闭或调用失败的回调函数                                                                |

### object.success 回调参数

| 属性               | 类型       | 说明                                             |
| ------------------ | ---------- | ------------------------------------------------ |
| cancel?            | boolean    | 为 true 时，表示取消                             |
| confirm?           | boolean    | 为 true 时，表示点击了确定，此时返回 `detail`    |
| detail?.value      | Date[]     | 开始 1 个 Date 实例 或 开始和结束 2 个 Date 实例 |
| detail?.valueArray | number[][] | 如：[[2021, 5, 1], [2021, 5, 30]]                |
| detail?.formatted  | string     | 05-21 ~ 05-22                                    |
| detail?.rangeCount | number     | 选择区间持续的天数（含首尾）                     |

### Usage

具体调用方式可以参考[API 调用](../guide/import.md#api-调用)。

```JavaScript
showCalendar({
  type: 'range',
  showClose: true,
  success: ({ confirm, cancel, detail }) => {
    ...
  }
})
```
