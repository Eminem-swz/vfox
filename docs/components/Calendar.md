# Calendar/CalendarPopup/CalendarView 日历

## Import

```
import { Calendar, CalendarPopup, CalendarView } from 'vfox'
```

具体的引入方式可以参考[引入组件](../guide/import.md)。

## 公共 Props

| 属性           | 类型                                                                                                     | 默认值             | 必填 | 说明                                                 |
| -------------- | -------------------------------------------------------------------------------------------------------- | ------------------ | ---- | ---------------------------------------------------- |
| v-model        | [SelectorValue](./Calendar.md#selectorvalue-的类型) \| [SelectorValue](./Calendar.md#selectorvalue-的类型)[] | []                 | 否   | 选中值                                               |
| min-date       | Date                                                                                                     | 当前日期           | 否   | 可选最小值                                           |
| max-date       | Date                                                                                                     | 当前日期的六个月后 | 否   | 可选最大值                                           |
| initial-mode   | [CalendarMode](./Calendar.md#calendarmode)                                                               | 'single'           | 否   | 模式                                                 |
| allow-same-day | boolean                                                                                                  | false              | 否   | `range` 模式生效，设置开始结束时间是否可以同一天     |
| max-range      | number                                                                                                   | Infinity           | 否   | `range` 模式生效，选择区间的最长天数                 |
| day-handler    | Function                                                                                                 |                    | 否   | 日历每个日期处理函数                                 |
| formatter      | [CalendarValueFormatter](./Calendar.md#calendarvalueformatter)                                           |                    | 否   | 和 `parser` 成对设置，对于 v-model 的值进行转化      |
| parser         | [CalendarValueParser](./Calendar.md#calendarvalueparser)                                                 |                    | 否   | 和 `formatter` 成对设置，对于 v-model 的值进行反转化 |

### SelectorValue 的类型

允许的类型为：`string | number | Date`

在不自定义 `formatter/parser` 的情况下，v-model 只有 `Date[]` 这种情况。

### SelectorModelValue 的类型

SelectorModelValue 的类型为： `SelectorValue | SelectorValue[]`

### CalendarDetail 的结构

| 值         | 类型                                                        | 说明                                                                                      |
| ---------- | ----------------------------------------------------------- | ----------------------------------------------------------------------------------------- |
| value      | [SelectorModelValue](./Calendar.md#selectormodelvalue-的类型) | 选择的值，`range` 模式下有开始 Date 和结束 Date 两个实例；`formatter/parser` 的优先级更高 |
| label      | string                                                      | 选中值对应的描述文本                                                                      |
| valueArray | number[][]                                                  | 如：[[2021, 5, 1], [2021, 5, 30]]                                                         |
| rangeCount | number                                                      | 选择区间持续的天数（含首尾）                                                              |

### CalendarMode

```
'single' | 'range'
```

`single` 表示选择一天，`range` 选择一个日期区间。

### dayHandler 函数

```
dayHandler(Object: Day) => Day
```

日历中的每个日期都对应一个 Day 对象，通过 `day-handler` 属性可以修改 Day 对象的内容后返回。

| 值              | 类型    | 说明                                                                                                                                                                                                                                         |
| --------------- | ------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| date            | Date    | 日期对应的 Date 对象，该字段修改无效                                                                                                                                                                                                         |
| state           | string  | 日期当前的状态，有：`selected` 选中，`startSelected` 开始（initialMode="range"下），`endSelected` 结束（initialMode="range"下），`disabled` 禁用。在空字符串的情况下，可以设置为 `disabled` 强制设置该日期为禁用（一般在票卖完的情况下设置） |
| text            | string  | 日期文本，不建议修改                                                                                                                                                                                                                         |
| topText         | string  | 日期上方展示文本，一般可修改为“节日名称”、“今天”、“明天”、“开始”、“结束”等                                                                                                                                                                   |
| topHighlight    | boolean | 日期上方展示文本是否高亮                                                                                                                                                                                                                     |
| bottomText      | string  | 日期下方展示文本，一般可修改为“机票价格”、“酒店价格”、“特价”、“热门”等                                                                                                                                                                       |
| bottomHighlight | boolean | 日期下方展示文本是否高亮                                                                                                                                                                                                                     |
| dateString      | string  | 日期对应的格式化时间（`YYYY-MM-DD`），如：2020-11-11，该字段不影响展示                                                                                                                                                                       |

### formatter/parser

高阶配置，`formatter` 和 `parser` 需要一同设置，对 v-model 的值转为自定义值。

#### CalendarValueFormatter

```
interface CalendarValueFormatter {
  (valueArray: Date[], mode: CalendarMode):
    | {
        value: SelectorValue | SelectorValue[]
        label: string
      }
    | (SelectorValue | SelectorValue[])
}
```

将 v-model 的原始值转为需要的自定义值，值需要满足 `SelectorValue | SelectorValue[]` 的类型约束，可以返回 { value, label } 对两个数据进行修改，或者单独返回 value。

#### CalendarValueParser

```
interface CalendarValueParser {
  (value: unknown, mode: CalendarMode): Date[]
}
```

跟 `CalendarValueFormatter` 相反，将自定义 v-model 的值转为组件认识的原始数组。

## Calendar

日历选择器。

注：

- 支持表单，具体可参考 [Form](./Form.md)。

## Calendar Props

| 属性         | 类型    | 默认值 | 必填 | 说明                                 |
| ------------ | ------- | ------ | ---- | ------------------------------------ |
| name         | string  |        | 否   | 标识                                 |
| placeholder  | string  |        | 否   | 没有选中值的提示，也会用在弹窗标题上 |
| disabled     | boolean | false  | 否   | 是否禁用                             |
| show-confirm | boolean | false  | 否   | 弹窗是否展示确定按钮                 |
| show-close   | boolean | false  | 否   | 弹窗是否展示关闭按钮                 |

## Calendar Events

| 事件   | 描述                   | 回调函数参数                                                | 函数 TypeScript  |
| ------ | ---------------------- | ----------------------------------------------------------- | ---------------- |
| change | 选择后值发生改变时触发 | [SelectorModelValue](./Calendar.md#selectormodelvalue-的类型) | SelectorOnChange |

## CalendarPopup

日历弹窗。

## CalendarPopup Props

| 属性            | 类型    | 默认值 | 必填 | 说明                   |
| --------------- | ------- | ------ | ---- | ---------------------- |
| v-model:visible | boolean | false  | 否   | 是否显示               |
| title           | string  |        | 否   | 弹窗标题               |
| show-confirm    | boolean | false  | 否   | 选择时是否展示确定按钮 |
| show-close      | boolean | false  | 否   | 是否展示关闭按钮       |

## CalendarPopup Events

| 事件                 | 描述                                              | 回调函数参数                                                 | 函数 TypeScript           |
| -------------------- | ------------------------------------------------- | ------------------------------------------------------------ | ------------------------- |
| confirm              | 选择完毕后触发 / `showConfirm` 点击确定按钮后触发 | [CalendarDetail](./Calendar.md#calendardetail-的结构)        | CalendarOnConfirm         |
| visible-state-change | 展示隐藏时触发                                    | { state: [VisibleState](./Calendar.md#visiblestate-值说明) } | PopupOnVisibleStateChange |

### VisibleState 值说明

| 值     | 说明                 | 备注                                              |
| ------ | -------------------- | ------------------------------------------------- |
| show   | 展示时触发           |                                                   |
| shown  | 展示且动画结束后触发 |                                                   |
| hide   | 隐藏时触发           | 可能携带其他参数 cancel, maskClick, closeClick 等 |
| hidden | 隐藏且动画结束后触发 | 可能携带其他参数 cancel, maskClick, closeClick 等 |

## CalendarView

日历选择，平铺形式。

## CalendarView Events

| 事件   | 描述       | 回调函数参数                                          | 函数 TypeScript  |
| ------ | ---------- | ----------------------------------------------------- | ---------------- |
| select | 选择后触发 | [CalendarDetail](./Calendar.md#calendardetail-的结构) | CalendarOnSelect |
