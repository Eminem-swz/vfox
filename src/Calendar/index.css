.fx-calendar {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  position: relative;
}
