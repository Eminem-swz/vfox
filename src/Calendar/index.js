import Calendar from './Calendar.vue'
import { showCalendar } from '../apis/Popup'

Calendar.install = function(Vue) {
  Vue.component(Calendar.name, Calendar)

  // 添加api
  Vue.prototype.$showCalendar = showCalendar
}

Calendar.showCalendar = showCalendar

export default Calendar
