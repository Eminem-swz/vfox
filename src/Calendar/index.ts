import { App } from 'vue'
import { SFCWithInstall } from '@/utils/types'
import Calendar from './Calendar.vue'
import { showCalendar } from '@/apis/Popup'

const _Calendar: SFCWithInstall<typeof Calendar> = Object.assign(Calendar, {
  install: function(app: App) {
    app.component(Calendar.name, Calendar)

    const properties = app.config.globalProperties

    properties.$showCalendar = showCalendar
  },
  showCalendar
})

export default _Calendar
