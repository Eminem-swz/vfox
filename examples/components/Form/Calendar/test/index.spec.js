/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Calendar from '@/components/Calendar'

describe('Calendar', () => {
  test('should render default correctly', () => {
    const wrapper = mount(Calendar, {})

    expect(wrapper.html()).toMatchSnapshot()
  })
})
