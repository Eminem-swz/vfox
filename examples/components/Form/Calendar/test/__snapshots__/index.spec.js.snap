// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar should render default correctly 1`] = `
<div class="fx-calendar">
  <div class="fx-input">
    <div class="fx-input_input placeholder"></div><svg class="fx-icon fx-input_arrow">
      <use xlink:href="#icon-RightOutlined"></use>
    </svg><input class="fx-input_cover fx-form-input" type="text" readonly="" name="">
  </div>
  <!--v-if-->
</div>
`;
